### ISSUE TITLE
Copyright headers missing from files

### ISSUE THREAD

**Issue by jpgr87** [flexible-collision-library/fcl#86]
The following files from the fcl-0.3.2 release contain no copyright headers:

```
# licensecheck -r .  | grep UNKNOW
./src/math/sampling.cpp: *No copyright* UNKNOWN
./src/continuous_collision.cpp: *No copyright* UNKNOWN
./include/fcl/exception.h: *No copyright* UNKNOWN
./include/fcl/math/vec_nf.h: *No copyright* UNKNOWN
./include/fcl/math/sampling.h: *No copyright* UNKNOWN
./include/fcl/ccd/simplex.h: UNKNOWN
./include/fcl/ccd/support.h: UNKNOWN
./include/fcl/continuous_collision.h: *No copyright* UNKNOWN
./include/fcl/learning/classifier.h: *No copyright* UNKNOWN
./test/libsvm_classifier.h: *No copyright* UNKNOWN
./test/general_test.cpp: *No copyright* UNKNOWN
./test/libsvm/svm.cpp: *No copyright* UNKNOWN
./test/libsvm/svm.h: *No copyright* UNKNOWN
./test/test_fcl_utility.cpp: *No copyright* UNKNOWN
./test/test_fcl_simple.cpp: *No copyright* UNKNOWN
```

Please consider adding proper copyright headers to the aforementioned files.

**Comment by scpeters** [comment]
The following files (added in https://github.com/flexible-collision-library/fcl/commit/bfc62b9b0d15026d8091952834d0536649b89ba0) have BSD license headers; I'm not sure why `licensecheck` doesn't see them:

* [include/fcl/ccd/simplex.h](https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/simplex.h)
* [include/fcl/ccd/support.h](https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/support.h)

The following file as migrated from svn (I think) by @isucan in 43ba254d172837682eb35a128c03489f27830273 without a license header:

* ./test/test_fcl_utility.cpp: *No copyright* UNKNOWN

The rest of the files were added without license headers by @panjia1983 in the following commits:

### de8666f68b9e190b448d059ad8ec4cc03761c3a5

* ./src/math/sampling.cpp: *No copyright* UNKNOWN
* ./include/fcl/exception.h: *No copyright* UNKNOWN
* ./include/fcl/learning/classifier.h: *No copyright* UNKNOWN

### 928b1756335451c61edf05903d65d99d9d74f632

* ./src/continuous_collision.cpp: *No copyright* UNKNOWN
* ./include/fcl/continuous_collision.h: *No copyright* UNKNOWN

### b16d32cf2840d84bde86df9613f18ecd8d0b7676

* ./include/fcl/math/vec_nf.h: *No copyright* UNKNOWN
* ./include/fcl/math/sampling.h: *No copyright* UNKNOWN

### 89887e9cdd398265686af070df602fa4a2b385a7

* ./test/general_test.cpp: *No copyright* UNKNOWN
* ./test/libsvm_classifier.h: *No copyright* UNKNOWN
* ./test/test_fcl_simple.cpp: *No copyright* UNKNOWN

### 6330ad3f6c2d8607ef0d77e360a22a2da37393ce

* ./test/libsvm/svm.cpp: *No copyright* UNKNOWN
* ./test/libsvm/svm.h: *No copyright* UNKNOWN

**Comment by sherm1** [comment]
Steve, I also have an earlier email from you about possible non-BSD code in FCL:
```
On Tue, Jun 16, 2015 at 11:48 PM, Steven Peters <scpeters@osrfoundation.org> wrote:
I was peeking at the FCL source code, and I think I remember Karen saying that there was
some non-open code from UNC in FCL. Grepping for UNC, I found comments in the following 
files that reference the CATCH library ( http://graphics.ewha.ac.kr/CATCH/ ), which requires 
a password to download. I don't know if there is directly copied code there or not.
```
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/interval.h#L36-L37
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/interval_matrix.h#L35-L36
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/interval_vector.h#L35-L36
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/taylor_matrix.h#L35-L36
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/taylor_model.h#L35-L36
https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/ccd/taylor_vector.h#L35-L36
```
Following links on the CATCH website, I also found the SWIFT++ library at UNC. This would be something to check as well.
http://gamma.cs.unc.edu/SWIFT++/download.html

Steve
```

It would be good to resolve these as well as the ones you mentioned here -- @panjia1983, can you help us figure out what are the correct licenses for this code?

**Comment by scpeters** [comment]
Good point @sherm1 

Those header files we discussed in the email have a BSD license listed, but the comment made we want to double-check that it was a correct license.

**Comment by mamoll** [comment]
I remember talking about this code with Stephane Redon at RSS2014. I think what happened is that some legacy code within UNC's Gamma lab made its way into FCL. Redon was a postdoc there, and so was Kim (now at EWHA). Jia was a graduate student there. Tracing the licensing / copyright is tricky to the “real” license / copyright holders is tricky.

**Comment by sherm1** [comment]
> Tracing the licensing / copyright to the “real” license / copyright holders is tricky.

I emailed Jia to see if he can weigh in. It occurred to me that he might not be getting notifications.

**Comment by jpgr87** [comment]
Any updates?

**Comment by jslee02** [comment]
#149 is created to move forward.

**Comment by jslee02** [comment]
Could we close this issue?

**Comment by jslee02** [comment]
I believe this is addressed by #149, please reopen if it's not.
